spec/rubocop/cop/style/redundant_regexp_character_class_spec.rb

```diff
@@ -43,6 +43,19 @@
     end
   end
 
+  context 'with a character class containing an escaped [' do
```

Comment:
This test wasn't strictly related to this PR, but I noticed we were missing it despite having the equivalent one for an escaped `]`